### PR TITLE
Synchronization Informations for imapsync

### PR DESCRIPTION
**Pull request by stephdl**
The PR displays some informations about the synchronisation (size, folder and messages remotely and locally)

https://github.com/NethServer/dev/issues/6299

This version implements a modal with rows

Screenshot (20)
Screenshot (19)
Screenshot (18)
Screenshot (21)

### CODE REVIEW

File: api/imapsync/count [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/perl
```
Comment (gsanchietti) [inline api/imapsync/count:1]: Please call this helper from the `read` api

**Review by gsanchietti — CHANGES_REQUESTED** [review]
I think we need to focus on user request: during job run, the admin would like to see the status of the current synchronization.
While all the information is present, data are difficult to read.

As you can see, there is no easy way for the user to know which is the synchronization percentage. I can propose to display the data in table-like manner or at least add a progress bar somewhere in the dialog.

Let's show some love for the user! ;)

**Comment by stephdl** [comment]
> As you can see, there is no easy way for the user to know which is the synchronization percentage. I can propose to display the data in table-like manner or at least add a progress bar somewhere in the dialog.

I thought about a progress bar like you can find elsewhere in the UI, but 100 000 emails to synchronize is not an important information, what is important too is the size 100 000 * 1KB (like for mailling list) of 100 000 * 10MB

So I think the progress bar is useless

**Comment by gsanchietti** [comment]
> So I think the progress bar is useless

Maybe a progress bar is not good, but a percentage of the size (or messages) already synchronized it's an helpful info.

> Probably what we could miss is the folder information because I believe they are created at the first sync

I agree.

File: ui/public/i18n/language.json [hunk 2]
```diff
@@ -484,7 +484,18 @@
...
         "imapsync_in_progress":"an IMAP synchronization is running",
         "task_in_progress":"Background task in progress",
-        "delete_imapsync_configuration":"Delete IMAP synchronization for "
+        "delete_imapsync_configuration":"Delete IMAP synchronization for ",
+        "Messages":"Messages",
+        "Sizes":"Sizes",
+        "Folders":"Folders",
+        "RemoteAccount":"Remote account",
+        "LocalAccount":"Local account",
+        "CheckAccountDiff":"Check account differences",
```
[the remark this answers is not in the document]
Comment (stephdl) [reply ui/public/i18n/language.json]: done

File: ui/public/i18n/language.json [hunk 3]
```diff
@@ -484,7 +484,18 @@
...
+        "PleaseWait":"Please wait, it might take a while",
```
[the remark this answers is not in the document]
Comment (stephdl) [reply ui/public/i18n/language.json]: done

**Comment by stephdl** [comment]
> Maybe a progress bar is not good, but a percentage of the size (or messages) already synchronized it's an helpful info.

A percentage, a gauge, a circular graphic  I think is a wrong way to display information to the sysadmin user. In some cases, the number of messages on the local server is greater than on the remote server.

- If you sync without deleting the local messages not in the remote server
- if you sync a remote account on an existing email account

With the cases above I met the point where I had a percentage > 100%, this will bring issues. Only the number and the total size of messages can bring important information and if something is wrong (Why I cannot the last X emails for example)

I suggest to go with the modal

**Comment by gsanchietti** [comment]
Let's go with the modal without the percentage

**Review by gsanchietti — APPROVED** [review]

**Comment by stephdl** [comment]
this is the modal we can see

Screenshot (22)

**Pull request merged**